Mobile/ShopLens/Droid/CameraActivity.cs

```diff
@@ -53,24 +62,34 @@ private bool IsThereAnAppToTakePictures()
         }
         private void TakeAPicture(object sender, EventArgs eventArgs)
         {
-            Intent intent = new Intent(MediaStore.ActionImageCapture);
-            _file = new File(_dir, string.Format("Image_{0}.jpg", Guid.NewGuid()));
-            intent.PutExtra(MediaStore.ExtraOutput, Uri.FromFile(_file));
-            StartActivityForResult(intent, 102);
+            Intent takePictureIntent = new Intent(MediaStore.ActionImageCapture);
+            productPhoto = new File(_dir, string.Format("Image_{0}.jpg", Guid.NewGuid()));
+            if (productPhoto != null)
+            {
+                Uri photoUri = FileProvider.GetUriForFile(ApplicationContext, ApplicationContext.PackageName + FILE_PROVIDER_NAME, productPhoto);
```

Comment:
We have to use a FileProvider to get access to SD card files.
You can read more about why this code is written so in the following links:
https://stackoverflow.com/questions/38200282/android-os-fileuriexposedexception-file-storage-emulated-0-test-txt-exposed# 
https://inthecheesefactory.com/blog/how-to-share-access-to-file-with-fileprovider-on-android-nougat/en